test/WebJobs.Script.Tests.Integration/Security/AuthenticationLevelHandlerTests.cs

```diff
@@ -0,0 +1,114 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.WebHost;
+using Microsoft.Azure.WebJobs.Script.WebHost.Authentication;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.Json;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+using Microsoft.Extensions.Hosting;
+using Microsoft.WebJobs.Script.Tests;
+using Moq;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace Microsoft.Azure.WebJobs.Script.Tests.Integration.Security
+{
+    [Trait(TestTraits.Category, TestTraits.EndToEnd)]
+    [Trait(TestTraits.Group, nameof(AuthenticationLevelHandlerTests))]
+    public class AuthenticationLevelHandlerTests
+    // TODO: This should be a more general E2E test suite - maybe for SWA? Or combine with storage/secret E3E tests?
```

Comment:
Agree that it would be good to have a dedicated test suite for the SWA scenario. I was experimenting myself, and sketched one out in branch https://github.com/Azure/azure-functions-host/tree/swa-fix. That might be a good starting point - it also addresses some of the issues I point out below.

Reply:
Transferred your test suite over. Did add the step of wiping existing test settings; the build machine has an `appsettings.test.json` that provides an `AzureWebJobsStorage` value otherwise. (See note in TestConfigurationBuilderExtensions.)